cmake/modules/zephyr_default.cmake

```diff
@@ -0,0 +1,89 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2021, Nordic Semiconductor ASA
+
+# This CMake module will load all Zephyr CMake modules in correct order for
+# default Zephyr build system.
+#
+# Outcome:
+# See individual CMake module descriptions
+
+include_guard(GLOBAL)
+
+# The code line below defines the real minimum supported CMake version.
+#
+# Unfortunately CMake requires the toplevel CMakeLists.txt file to define the
+# required version, not even invoking it from a CMake module is sufficient.
+# It is however permitted to have multiple invocations of cmake_minimum_required.
+cmake_minimum_required(VERSION 3.20.0)
+
+message(STATUS "Application: ${APPLICATION_SOURCE_DIR}")
+
+find_package(ZephyrBuildConfiguration
+  QUIET NO_POLICY_SCOPE
+  NAMES ZephyrBuild
+  PATHS ${ZEPHYR_BASE}/../*
+  NO_CMAKE_PATH
+  NO_CMAKE_ENVIRONMENT_PATH
+  NO_SYSTEM_ENVIRONMENT_PATH
+  NO_CMAKE_PACKAGE_REGISTRY
+  NO_CMAKE_SYSTEM_PATH
+  NO_CMAKE_SYSTEM_PACKAGE_REGISTRY
+)
+
+# Prepare user cache
```

Comment:
This comment should be dropped.

We are not preparing the user cache here.

Reply:
we prepare in the sense that we lookup and determine the directory to use, including a test to ensure it is writable.
This guarantees the user cache, is ready when needed.

Feel free to propose a different comment, but I believe `prepare` is correct in this case.

Reply:
We are finding python and configuring ccache here. We are not finding the user_cache here, that happens later.

Reply:
Nicely spotted.

As discussed here https://github.com/zephyrproject-rtos/zephyr/pull/41301#discussion_r786720876 user_cache is a dependency of extension, and therefore it should also be loaded first. 

accidentally I moved ccache instead of user_cache, but the comment was correct.
This is now fixed:
https://github.com/zephyrproject-rtos/zephyr/blob/542b99f0dc53dd489a2240dfd4e4d683830fbb22/cmake/modules/zephyr_default.cmake#L34-L36